crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -97,9 +99,17 @@ impl StructInfo {
             type_id: TypeId::of::<T>(),
             fields: fields.to_vec().into_boxed_slice(),
             field_indices,
+            #[cfg(feature = "documentation")]
+            doc: None,
         }
     }
 
+    /// Sets the docstring for this struct.
+    #[cfg(feature = "documentation")]
+    pub fn with_doc(self, doc: Option<&'static str>) -> Self {
+        Self { doc, ..self }
+    }
```

Comment:
For reference, the reason I went with the builder-like method instead of a second constructor is that I felt it set a better example for future contributors. Otherwise, if we ever add more feature-gated data to type info, then we'd end up with one or more unwieldy constructors.

If we'd prefer to make this it's own constructor, then I can change all instances.